project-koku/koku-metrics-operator
# COST-547 - introduce configurable PVC

- PVC is configurable thru the CR
- If no PVC config is provided, a default is used
- reports directory is initialized in deployment as EmptyDir
- EmptyDir is replaced with PVC
- The Pod reboots after deployment modification, and new Pod access the new PVC

Testing:
1. deploy with the default CR and ensure that the built-in default PVC is created and mounted.
2. keep the PVC from previous step, and redeploy the operator. Create a new CR and ensure that the deployment is mounted onto the previously created PVC (if reports were created in step 1, you could check that they still exist - this test is equivalent to what could happen during an operator upgrade)
3. re-deploy, create a new CR with a defined PVC. Ensure that the specked PVC is created and mounted.
4. create a new PVC. re-deploy. Create a CR with the created PVC spec. Ensure that deployment is mounted onto that created PVC
5. continuing with previous deployment, remove the PVC definition from the CR. Ensure the deployment is re-mounted onto the built-in default PVC. Also check to make sure that the PVC that was unmounted remains within the project
